### PR TITLE
Issue #49 delay import of pandas 

### PR DESCRIPTION
**Pull request by jmrgibson**
Should speed the loading of this plugin up a bit, for those that have pandas installed but were not using it

### CODE REVIEW

File: pytest_harvest/plugin.py [hunk 1]
```diff
@@ -219,62 +219,60 @@ def module_results_dct(request, fixture_store):
...
-                               fixture_store=FIXTURE_STORE,            # type: Union[Mapping[str, Any], Iterable[Mapping[str, Any]]]
-                               results_bag_fixture_name='results_bag'  # type: str
-                               ):
-        # type: (...) -> pd.DataFrame
-        """
-        Helper method to get exactly the same object than the `session_results_df` fixture, from a session object.
+def get_session_results_df(session_or_request,
+                            fixture_store=FIXTURE_STORE,            # type: Union[Mapping[str, Any], Iterable[Mapping[str, Any]]]
+                            results_bag_fixture_name='results_bag'  # type: str
+                            ):
```
Comment (smarie) [inline pytest_harvest/plugin.py:225]:
```suggestion
                           fixture_store=FIXTURE_STORE,            # type: Union[Mapping[str, Any], Iterable[Mapping[str, Any]]]
                           results_bag_fixture_name='results_bag'  # type: str
                           ):
```

File: pytest_harvest/plugin.py [hunk 2]
```diff
@@ -219,62 +219,60 @@ def module_results_dct(request, fixture_store):
...
+    :param session_or_request: the pytest session or request
+    :param fixture_store: an optional fixture store
+    :param results_bag_fixture_name: an optional name for results bag fixture in the fixture store. Default is
+        "results_bag"
+    :return:
+    """
+    try:
+        import pandas as pd  # pylint: disable=import-outside-toplevel
+    except ImportError as e:
+        saved_e = e
```
Comment (smarie) [inline pytest_harvest/plugin.py:242]:
```suggestion
```

File: pytest_harvest/plugin.py [hunk 3]
```diff
@@ -292,64 +290,62 @@ def session_results_df(request, fixture_store):
...
+    :param filter: any filter, see `get_session_synthesis_dct` for details
+    :param fixture_store: an optional fixture store
+    :param results_bag_fixture_name: an optional name for results bag fixture in the fixture store. Default is
+        "results_bag"
+    :return:
+    """
+    try:
+        import pandas as pd  # pylint: disable=import-outside-toplevel
+    except ImportError as e:
+        saved_e = e
```
Comment (smarie) [inline pytest_harvest/plugin.py:313]:
```suggestion
```

File: pytest_harvest/plugin.py [hunk 4]
```diff
@@ -292,64 +290,62 @@ def session_results_df(request, fixture_store):
...
 
+    # in case of xdist, make sure persisted workers results have been reloaded
+    possibly_restore_xdist_workers_structs(session)
+
+    # get the synthesis dictionary, merged with default fixture store and flattening default results_bag
+    module_results_dct = get_session_synthesis_dct(session, durations_in_ms=True,
+                                                    filter=filter,
+                                                    test_id_format=test_id_format, status_details=False,
+                                                    fixture_store=fixture_store,
+                                                    flatten=True, flatten_more=results_bag_fixture_name)
```
Comment (smarie) [inline pytest_harvest/plugin.py:326]:
```suggestion
                                                   filter=filter,
                                                   test_id_format=test_id_format, status_details=False,
                                                   fixture_store=fixture_store,
                                                   flatten=True, flatten_more=results_bag_fixture_name)
```

File: pytest_harvest/plugin.py [hunk 5]
```diff
@@ -292,64 +290,62 @@ def session_results_df(request, fixture_store):
...
+        "results_bag"
+    :return:
+    """
+    try:
+        import pandas as pd  # pylint: disable=import-outside-toplevel
+    except ImportError as e:
+        saved_e = e
         six.raise_from(Exception("There was an error importing `pandas` module. Fixture `session_results_df` and "
                                  "methods `get_filtered_results_df` and `get_module_results_df` can not be used in this"
                                  " session. "), saved_e)
```
Comment (smarie) [inline pytest_harvest/plugin.py:316]:
```suggestion
                                 " session. "), e)
```

**Comment by smarie** [comment]
Merged, thanks very much !

**Comment by jmrgibson** [comment]
Thanks for the quick turnaround!

**Pull request merged**